### PR TITLE
feat: added new docs for spacing in design language

### PR DESCRIPTION
**Pull request by yorbi-dp**
Jira: https://dialpad.atlassian.net/browse/DLT-1307

## Description
- Added documentation for Spacing and Sizing views in the Design Language section.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

**Comment by iropolo** [comment]
visual-test will not work for projects that are not vue3, we need to update the script

**Review by iropolo — APPROVED** [review]
if its looks good and have design approvement, ready to go 🚀

File: packages/dialtone/lib/build/less/components/table.less
Comment (francisrupert) [inline packages/dialtone/lib/build/less/components/table.less]: What's this about? This is a global change to the Table style and should probably remain `200`

**Review by francisrupert — CHANGES_REQUESTED** [review]
This is looking awesome! I'll get to this, especially for content.

**Comment by juliodialpad** [comment]
> visual-test will not work for projects that are not vue3, we need to update the script

Just need to update the branch, the change is already there.

@yorbi-dp pull the latest changes from staging please.

File: packages/dialtone/lib/build/less/components/table.less
Comment (yorbi-dp) [reply packages/dialtone/lib/build/less/components/table.less]: Oh my bad! Just fixed it in the new commit.

File: apps/dialtone-documentation/docs/design/sizing/index.md
Comment (braddialpad) [inline apps/dialtone-documentation/docs/design/sizing/index.md]: `var(--dt-size-300)` rather than space-300

File: apps/dialtone-documentation/docs/design/sizing/index.md
Comment (braddialpad) [inline apps/dialtone-documentation/docs/design/sizing/index.md]: small grammatical error here change to either: "when setting a minimum dimension" or "when setting minimum dimension"

Comment (braddialpad) [inline apps/dialtone-documentation/docs/design/sizing/index.md]: I'm a bit confused by this paragraph, it talks about not setting fixed width / height but then below states dialtone components that DO have fixed width height. Are we saying that these components are exceptions to this rule?

It's also unclear whether we are promoting the usage of min-width / min-height. Are we promoting the use of min-width / min-height as much as possible for greater responsiveness? or are we saying that we should avoid using min-width / min-height unless you absolutely need to use it?

File: apps/dialtone-documentation/docs/design/sizing/index.md
Comment (braddialpad) [inline apps/dialtone-documentation/docs/design/sizing/index.md]: We state that using predefined tokens is one of the key takeaways of this page, however we never really mention anything about the usage of tokens, we just show a list of them.

File: apps/dialtone-documentation/docs/design/spacing/index.md
Comment (braddialpad) [inline apps/dialtone-documentation/docs/design/spacing/index.md]: Did we mention anything about this in the document?

**Review by braddialpad — COMMENTED** [review]
Sorry for the delay. Couple comments on wording and such. Feel free to merge and make these changes separately if we are looking to get this out quickly.

File: apps/dialtone-documentation/docs/design/sizing/index.md
Comment (yorbi-dp) [reply apps/dialtone-documentation/docs/design/sizing/index.md]: Thanks for pointing that out! Fixed it in the new commit.

File: apps/dialtone-documentation/docs/design/sizing/index.md
Comment (yorbi-dp) [reply apps/dialtone-documentation/docs/design/sizing/index.md]: Oh, just added more info about this in the latest commit. Thank you! 🙌

File: apps/dialtone-documentation/docs/design/spacing/index.md
Comment (yorbi-dp) [reply apps/dialtone-documentation/docs/design/spacing/index.md]: Updated in the latest commit. Thank you!

**Review by braddialpad — APPROVED** [review]

**Comment by juliodialpad** [comment]
Why is this PR not merged still? are we missing some approval or changes?

**Comment by braddialpad** [comment]
Not sure.. @yorbi-dp this good to go?

**Comment by yorbi-dp** [comment]
We're waiting on final approvals before merging. 🙌  Should I convert it to draft?

**Comment by braddialpad** [comment]
Nah it's okay. Just wanted to know what's going on with it since it's been approved for 5 days with no activity. If we're still waiting on approvals from design its fine to leave in this state.

File: apps/dialtone-documentation/docs/design/sizing/index.md
Comment (francisrupert) [inline apps/dialtone-documentation/docs/design/sizing/index.md]: On both the Size and the Spacing page, let's introduce the difference between them so designers and engineers can understand the difference and why they are distinct.

In the product codebase I too often see them conflated; e.g...
```less
// 👎 Incorrect
padding: var(--dt-size-400)`;

// 👍 Correct
padding: var(--dt-space-400)`;
```

It needs to be clear at this introductory doc level which they should be choose. Below is a _very_ rough example of how it can be introduced. Think about how what I've drafted below can be used in both, perhaps in a visual way: 

```md
## Choosing Size vs Spacing design tokens

### Size
Size design tokens define intrinsic **dimensions** of UI
components. Example CSS properties they correspond to:
- `width`
- `height`
- `border-radius`
- `border-width`
- positioning properties, i.e. `top`, `left`, etc.

### Spacing
Spacing design tokens focus on controlling **spatial 
relationships**; that is, the space between and around
elements. Example CSS properties they correspond to:
- `padding`
- `margin`
- `gap`
```

**Review by francisrupert — CHANGES_REQUESTED** [review]
I'd say this is pretty much there. These pages will be a big hit. Just one key change will drive it home by introducing the difference between Size and Spacing.

File: apps/dialtone-documentation/docs/design/size/index.md
Comment (francisrupert) [inline apps/dialtone-documentation/docs/design/size/index.md]: For both Size and Space, use the Info variant of [Notice](https://dialtone.dialpad.com/components/notice.html#base-styles) for this line.

**Review by francisrupert — APPROVED** [review]
Awesome sauce! Just one little change and then **ship it!**

File: apps/dialtone-documentation/docs/design/size/index.md
Comment (francisrupert) [reply apps/dialtone-documentation/docs/design/size/index.md]: (Just the "Note:" part, now the above "WIP" part.

Comment (yorbi-dp) [reply apps/dialtone-documentation/docs/design/size/index.md]: Great! Thank youuu

**Pull request merged**
